test/OpenTelemetry.Tests/Trace/TracerTest.cs

```diff
@@ -74,6 +76,35 @@ public void Tracer_StartRootSpan_BadArgs_NullSpanName()
             Assert.Null(span3.Activity.DisplayName);
         }
 
+        [Fact(Skip = "See https://github.com/open-telemetry/opentelemetry-dotnet/issues/2803")]
+        public async Task Tracer_StartRootSpan_StartsNewTrace()
+        {
+            var exportedItems = new List<Activity>();
+
+            using var tracer = Sdk.CreateTracerProviderBuilder()
+                .AddSource("tracername")
+                .AddInMemoryExporter(exportedItems)
+                .Build();
+
+            async Task DoSomeAsyncWork()
+            {
+                await Task.Delay(10);
+                var newRootSpan = TracerProvider.Default.GetTracer("tracername").StartRootSpan("RootSpan2");
+                using (Tracer.WithSpan(newRootSpan))
+                {
+                    await Task.Delay(10);
+                }
+            }
+
+            using (tracer.GetTracer("tracername").StartActiveSpan("RootSpan1"))
+            {
+                await DoSomeAsyncWork();
+            }
+
+            Assert.Equal(2, exportedItems.Count);
+            Assert.NotEqual(exportedItems[0].TraceId, exportedItems[1].TraceId);
```

Comment:
would be good to make it clear in this line which one is item0 and item1, by doing an assert of the `DisplayName`.